### ISSUE TITLE
Getting thrown out of my desktop session every day or so

### ISSUE THREAD

**Issue by sfatula** [raspberrypi/bookworm-feedback#184]
I'll be sitting here on my Pi 5 kernel 6.1.0-rpi7-rpi-2712 and all of the sudden I am logged out and anything I was working on and not saved, gone of course. Just starting to load my new PI 5, so, not much loaded yet. Here's the log file from journalctl from the latest time I was thrown out of my session. Not sure what to do here but suspect a bug. 

https://pastebin.com/WJRMpPxk

**Comment by lurch** [comment]
Possibly the same thing as #146 ?

**Comment by sfatula** [comment]
Yeah I read that but not sure it's the same. The OP talks about monitor being switched off. I am sitting here typing as it happens with my monitor on (of course). It also talks about pygame a lot, not using it. The log file looks different, not the same messages. 

Now, I do turn off my monitor each night and it could well be that causes my nightly issue, I do have that as each morning I am greeted with a login screen. But it's also happening while using the machine which is different I think. Strangely, I noted the time it happened today (when first using machine) and it was about exactly when I turned the monitor on. 

I would try the vc4.force_hotplug though I am not sure it's still valid on the PI 5 with bookworm? 

It should be noted I am current on all the software updates.

**Comment by lurch** [comment]
> I would try the vc4.force_hotplug though I am not sure it's still valid on the PI 5 with bookworm?

It is, see my comments here https://github.com/raspberrypi/bookworm-feedback/issues/67#issuecomment-1863340591

**Comment by sfatula** [comment]
So, adding that to config.txt option made no difference as far as the during my use portion of the issue. i.e., was typing away tonight and in the middle of typing was logged out (or it crashed, whatever) and had to log back in. Will see in the morning if it affects the turning the monitor off portion.

**Comment by 6by9** [comment]
> So, adding that to config.txt option 

It's an option for cmdline.txt, not config.txt

**Comment by sfatula** [comment]
Yep, typed the wrong thing, I added it to cmdline.txt, sorry about that. I did wake up today to a machine that did not have the login window, so, it survived all night. Adding it may have resolved the monitor off all night problem. Doesn't help the during use problem though. 

My cmdline.txt looks like this:

console=serial0,115200 console=tty1 root=PARTUUID=cfd930c2-02 rootfstype=ext4 fsck.repair=yes rootwait quiet splash plymouth.ignore-serial-consoles cfg80211.ieee80211_regdom=US vc4.force_hotplug=2

**Comment by popcornmix** [comment]
> vc4.force_hotplug=2

so you have one monitor attached and it's connected to HDMI-A-2 (rather than HDMI-A-1)?
It would be more usual to use the other socket (closest to power connector) if only using one monitor.

**Comment by sfatula** [comment]
Correct, am using the furthest away. Can I not use it? It makes it easier to plug in a second one if/when I want to as I can then see the place to plug it in.

**Comment by popcornmix** [comment]
Some fullscreen apps (like kodi in fullscreen mode) only support the first hdmi display.
And most help given about configuring displays will assume you are using the first one.

But if you are happy with that, then it should be fine generally.

On Pi4, the recommendation is slightly stronger, as the hardware for the first hdmi port is slightly more capable (it can support higher pixel rates), but on Pi5 the hardware is the same.

**Comment by sfatula** [comment]
Still getting thrown out daily, just never with monitor off. I even swapped HDMI cables just in case.  I also switched to the first video port. No change. I found some discussion of this same issue in the Pi forums:

https://forums.raspberrypi.com/viewtopic.php?t=357538&start=875#p2165075
